File: .temp_patch_datingservice.js
const fs = require('fs');
const path = require('path');
const file = path.join('src', 'services', 'datingService.ts');
let text = fs.readFileSync(file, 'utf8');
text = text.replace(/const \{ data: \{ user \} \} = await supabase\.auth\.getSession\(\);/g, 'const { data } = await supabase.auth.getSession(); const user = data.session?.user;');
text = text.replace(/msg => msg\.sender_id !== user\.id && !msg\.read_at/g, '(msg: any) => msg.sender_id !== user.id && !msg.read_at');
const oldProfileReturn = `      return {
      id: data.id,
      name: data.display_name || 'Anonymous',
      age: data.age || 25,
      location: data.location || 'Bangkok',
      photos: data.dating_photos || [],
      bio: data.bio || '',
      relationshipGoal: data.relationship_goals || '',
      occupation: data.occupation,
      education: data.education,
      mbti: data.mbti,
      loveLanguage: data.love_language,
    };
`;
const newProfileReturn = `      return {
      id: data.id,
      name: data.display_name || 'Anonymous',
      age: data.age || 25,
      location: data.location || 'Bangkok',
      photos: data.dating_photos || [],
      compatibility: 0,
      lifestyleScore: 0,
      personalityScore: 0,
      emotionalScore: 0,
      communicationScore: 0,
      insights: [],
      interests: [],
      bio: data.bio || '',
      relationshipGoal: data.relationship_goals || '',
      occupation: data.occupation,
      education: data.education,
      mbti: data.mbti,
      loveLanguage: data.love_language,
    };
`;
if (text.includes(oldProfileReturn)) {
  text = text.replace(oldProfileReturn, newProfileReturn);
} else {
  console.log('did not find getMyProfile return pattern');
}
const oldMatchReturn = `    return {
      id: match.id,
      partnerId: partner.id,
      partnerName: partner.display_name || 'Anonymous',
      partnerAvatar: partner.avatar_url || '',
      compatibilityScore: match.compatibility_score || 0,
      isUltraMatch: match.is_ultra_match || false,
      hasConversation: match.has_conversation || false,
      createdAt: new Date(match.created_at),
    };
`;
const newMatchReturn = `    return {
      id: match.id,
      partnerId: partner.id,
      partnerName: partner.display_name || 'Anonymous',
      partnerAvatar: partner.avatar_url || '',
      compatibilityScore: match.compatibility_score || 0,
      isUltraMatch: match.is_ultra_match || false,
      lastMessage: undefined,
      lastMessageTime: undefined,
      unreadCount: 0,
      hasConversation: match.has_conversation || false,
      createdAt: new Date(match.created_at),
    };
`;
if (text.includes(oldMatchReturn)) {
  text = text.replace(oldMatchReturn, newMatchReturn);
} else {
  console.log('did not find getMatch return pattern');
}
fs.writeFileSync(file, text, 'utf8');
console.log('patched datingService.ts');

File: .temp_patch_datingservice.py
from pathlib import Path
path = Path('src/services/datingService.ts')
text = path.read_text(encoding='utf8')
text = text.replace("const { data: { user } } = await supabase.auth.getSession();", "const { data } = await supabase.auth.getSession(); const user = data.session?.user;")
text = text.replace("msg => msg.sender_id !== user.id && !msg.read_at", "(msg: any) => msg.sender_id !== user.id && !msg.read_at")
old = "      return {\n      id: data.id,\n      name: data.display_name || 'Anonymous',\n      age: data.age || 25,\n      location: data.location || 'Bangkok',\n      photos: data.dating_photos || [],\n      bio: data.bio || '',\n      relationshipGoal: data.relationship_goals || '',\n      occupation: data.occupation,\n      education: data.education,\n      mbti: data.mbti,\n      loveLanguage: data.love_language,\n    };\n"
new = "      return {\n      id: data.id,\n      name: data.display_name || 'Anonymous',\n      age: data.age || 25,\n      location: data.location || 'Bangkok',\n      photos: data.dating_photos || [],\n      compatibility: 0,\n      lifestyleScore: 0,\n      personalityScore: 0,\n      emotionalScore: 0,\n      communicationScore: 0,\n      insights: [],\n      interests: [],\n      bio: data.bio || '',\n      relationshipGoal: data.relationship_goals || '',\n      occupation: data.occupation,\n      education: data.education,\n      mbti: data.mbti,\n      loveLanguage: data.love_language,\n    };\n"
if old in text:
    text = text.replace(old, new)
else:
    print('did not find getMyProfile return pattern')
old2 = "    return {\n      id: match.id,\n      partnerId: partner.id,\n      partnerName: partner.display_name || 'Anonymous',\n      partnerAvatar: partner.avatar_url || '',\n      compatibilityScore: match.compatibility_score || 0,\n      isUltraMatch: match.is_ultra_match || false,\n      hasConversation: match.has_conversation || false,\n      createdAt: new Date(match.created_at),\n    };\n"
new2 = "    return {\n      id: match.id,\n      partnerId: partner.id,\n      partnerName: partner.display_name || 'Anonymous',\n      partnerAvatar: partner.avatar_url || '',\n      compatibilityScore: match.compatibility_score || 0,\n      isUltraMatch: match.is_ultra_match || false,\n      lastMessage: undefined,\n      lastMessageTime: undefined,\n      unreadCount: 0,\n      hasConversation: match.has_conversation || false,\n      createdAt: new Date(match.created_at),\n    };\n"
if old2 in text:
    text = text.replace(old2, new2)
else:
    print('did not find getMatch return pattern')
path.write_text(text, encoding='utf8')
print('patched datingService.ts')
